code/parachain/frame/cosmwasm/src/mock.rs

```diff
@@ -197,29 +197,36 @@ impl pallet_timestamp::Config for Test {
 
 /// Native <-> Cosmwasm account mapping
 pub struct AccountToAddr;
+
 impl Convert<alloc::string::String, Result<AccountId, ()>> for AccountToAddr {
 	fn convert(a: alloc::string::String) -> Result<AccountId, ()> {
-		match a.strip_prefix("0x") {
-			Some(account_id) => Ok(<[u8; 32]>::try_from(hex::decode(account_id).map_err(|_| ())?)
-				.map_err(|_| ())?
-				.into()),
-			_ => Err(()),
-		}
+		let account =
+			ibc_primitives::runtime_interface::ss58_to_account_id_32(&a).map_err(|_| ())?;
```

Comment:
not sure, error is ().  may you suggest why not even String?

Reply:
Error can only be `SS58CodecError::InvalidString` or `InvalidAccountId`. `pallet-cosmwasm` raises `AccountConversionFailure` anyways, so catching the error is not necessary here.